src/cli/generator/config.rs

```diff
@@ -90,9 +91,17 @@ pub struct Input<Status = UnResolved> {
 #[derive(Deserialize, Serialize, Debug)]
 #[serde(rename_all = "camelCase")]
 pub enum Source<Status = UnResolved> {
-    Curl { src: Location<Status> },
-    Proto { src: Location<Status> },
-    Config { src: Location<Status> },
+    Curl {
+        src: Location<Status>,
+        #[serde(skip_serializing_if = "Option::is_none")]
```

Comment:
probabably could be written as 
```suggestion
        #[serde(skip_serializing_if = "is_default")]
```